src/tests/evaluate.test.ts

```diff
@@ -3,27 +3,163 @@ import { evaluate } from '../model/evaluator/evaluate';
 import { parse } from '../model/parser';
 
 describe('simple query permutation tests', () => {
-  it('generates correct query permutations for "p and q"', () => {
-    const expectedResult: QueryPermutation[] = [
-      {
-        queryParameters: new Map<string, boolean>([['p', false], ['q', false]]),
-        value: false,
-      },
-      {
-        queryParameters: new Map<string, boolean>([['p', false], ['q', true]]),
-        value: false,
-      },
-      {
-        queryParameters: new Map<string, boolean>([['p', true], ['q', false]]),
-        value: false,
-      },
-      {
-        queryParameters: new Map<string, boolean>([['p', true], ['q', true]]),
-        value: true,
-      },
-    ];
-
-    expect(evaluate(parse('p and q'))).toStrictEqual(expectedResult);
+  const testCases: {expression: string, truthTable: any[]}[] = [
```

Comment:
kind of a nit, but ['p', 'q'] aren't really part of the truth table. why not just add another property that's like 'variableNames'? and then it could be type safe too

Reply:
Definitely a good point